### DISCUSSION TITLE
Unable to connect to rustdesk on many networks

### DISCUSSION THREAD

**Discussion by rohitsingal1** [rustdesk/rustdesk#4900] [Q&A]
I have a self hosted rustdesk and it's an amazing software. 
At my office I have 10 client systems which are using my self hosted "ID Server" and "Key"
My server is on another network (Separate connection and IP). 
My issue is that I get "Connection in progress" screen when I connect to rustdesk through other networks and no display.
After few seconds it gives an error message "Failed to connect via relay server: Failed to connect to relay server: Please try later"

For example if I use a hotel/airport network I get "Connection in progress" and then "Failed to connect via relay server..........".
If I use my phone's hotspot I am able to connect but using my colleagues phone hotspot I get the error.
In fact on some mobile hotspot it works, some don't and same in the case of hotel/airports. 

The weird thing is if the client is online it just shows "Connection in progress" error but if it is offline  it immediately says "Remote Desktop is offline".

I have been through Issue 24 https://github.com/rustdesk/rustdesk-server/issues/24. I am already using xyz.abc.com as my relay server address.

**Comment by DevJupyHUB** [comment]
_@rohitsingal1 "For example if I use a hotel/airport network I get "Connection in progress" and then "Failed to connect via relay server.........."."_


Hi there,

Ports used by RustDesk are blocked in many public places with f.e. public WiFi (internet cafes, airports, etc.)
This is a system security consideration from the network operator and not a RustDesk issue...


these are the ports we are talking about:

21115:21119/tcp
21116/udp

**Comment by rohitsingal1** [comment]
@DevJupyHUB 
I am not that much of a technical guy but shouldn't the port which is receiving the request be open in case my server where I have opened firewall ports 21115 to 21119 and also on my router.

Also the hotspot of mobile networks ? What about them ?

**Comment by DevJupyHUB** [comment]
@rohitsingal1 "Also the hotspot of mobile networks ? What about them ?"

Your server is not the problem here, if ports are open correctly...
f.e.:
ufw allow 21115:21119/tcp
ufw allow 21116/udp
sudo ufw enable

The problem is on the network you're trying to connect from, because the firewall there is blocked on those ports, so you can't get out of the network...

BTW:
if you want to check this pls. NMAP to your server and check your ports 

from laptop: https://nmap.org/download
from phone: https://secwiki.org/w/Nmap/Android

+++edit:
_even simpler:_

For security reasons, on a public (free) network, most of the time **only ports 80/443 (http/https) are opened** to allow you to connect to the Internet, but all other ports are blocked
(like a captive portal)

**Comment by rohitsingal1** [comment]
Thanks @DevJupyHUB 
It will take me sometime to understand the links you shared.

This issue would be faced by multiple users then.

Also is it possible to run rustdesk on port 80  through nginx as I already have 2 webapps running ?

**Comment by DevJupyHUB** [comment]
@rohitsingal1 _"This issue would be faced by multiple users then."_

Yes this has been a topic here, if you search a bit you will find it.

@rohitsingal1 _"run rustdesk on port 80_"
There will be or already is a web version:
https://rustdesk.com/docs/en/self-host/pro/console/


@rohitsingal1 _"nginx as I already have 2 webapps running ?"_
The Nginx,..... has been tried by several people with some success, but I think it's just an extra failure option, so I personally don't prefer it

++edit:
I personally never connect to a public network, what I do  (when I am on the move and/or on the road) 4 or 5G plus VPN and everything stays safe

**Comment by rohitsingal1** [comment]
@DevJupyHUB 
Thanks for your time.
I did search and find this https://github.com/rustdesk/rustdesk/discussions/984.
It's not supported as of now.
Will close this with this final comment.

**Comment by DevJupyHUB** [comment]
@rohitsingal1 "Thanks for your time."
You're welcome :-)
